app/models/manageiq/providers/ibm_cloud/vpc/network_manager/cloud_subnet.rb

```diff
@@ -14,6 +15,49 @@ class ManageIQ::Providers::IbmCloud::VPC::NetworkManager::CloudSubnet < ::CloudS
     end
   end
 
+  def self.params_for_create(ems)
+    {
+      :fields => [
+        {
+          :component    => 'select',
+          :name         => 'cloud_network_id',
+          :id           => 'cloud_network_id',
+          :label        => _('Network'),
+          :isRequired   => true,
+          :includeEmpty => true,
+          :validate     => [{:type => 'required'}],
+          :options      => ems.cloud_networks.map do |cvt|
+            {
+              :label => cvt.name,
+              :value => cvt.id,
+            }
+          end
+        }
+      ]
+    }
+  end
+
+  def self.raw_create_cloud_subnet(ext_management_system, options)
+    cloud_network_id = options.delete(:cloud_network_id)
```

Comment:
Minor but since you aren't passing `options` directly into `connection.request(:create_subnet)` you don't need to delete the key here.

Some other providers (like openstack) pass the `options` has directly into their API call and so have to delete any keys which aren't expected.

You can just use `options[:cloud_network_id]`